presto-main/src/main/java/com/facebook/presto/type/LikeFunctions.java

```diff
@@ -94,6 +96,63 @@ public static Regex likePattern(@SqlType("varchar(x)") Slice pattern, @SqlType("
         return likePattern(pattern.toStringUtf8(), getEscapeChar(escape), true);
     }
 
+    public static boolean isLikePattern(Slice pattern, Slice escape)
+    {
+        String stringPattern = pattern.toStringUtf8();
+        if (escape == null) {
+            return stringPattern.contains("%") || stringPattern.contains("_");
+        }
+
+        String stringEscape = escape.toStringUtf8();
+        checkCondition(stringEscape.length() == 1, INVALID_FUNCTION_ARGUMENT, "Escape must be empty or a single character");
+
+        char escapeChar = stringEscape.charAt(0);
+        boolean escaped = false;
+        boolean isLikePattern = false;
+        for (int currentChar : stringPattern.codePoints().toArray()) {
+            if (!escaped && (currentChar == escapeChar)) {
+                escaped = true;
+            }
+            else if (escaped) {
+                checkEscape(currentChar == '%' || currentChar == '_' || currentChar == escapeChar);
+                escaped = false;
+            }
+            else if ((currentChar == '%') || (currentChar == '_')) {
+                isLikePattern = true;
```

Comment:
Both ways can make sense to me. An invalid like pattern is not really a like  pattern, so it makes sense to check whether it's a valid pattern here to me. But I also see a point to separate "checkLikePattern" and "isLikePatternAPattern". I have no strong opinions.